app/src/main/java/com/startup/go4lunch/ui/RestaurantDetailActivityViewModel.java

```diff
@@ -25,23 +26,34 @@ public RestaurantDetailActivityViewModel(@NonNull RestaurantRepository restauran
         this.workmateRepository = workmateRepository;
     }
 
-    @Nullable
-    public Restaurant getRestaurantFromId(long idRestaurant) {
-        return restaurantRepository.getRestaurantFromId(idRestaurant);
+    @NonNull
```

Comment:
Par extension (O du SOLID), il serait préférable de créer filtrer la liste reçue en paramètre, par ID plutôt que de faire un nouvel appel au repository.

Si le getRestaurantFromId donne plus d'info que les restaus renvoyés dans le live data, renommer la méthode en `getRestaurantFromId` en `getDetailsFor` par example

Reply:
Pour le filtre, tu peux itérer sur la liste et vérifier l'ID, retourner null sinon. 

Une façon compacte de le faire en java est de transformer la liste en stream et de trouver le premier qui match l'ID

```java
Transformations.map(
    restaurantRepository.getRestaurantListLiveData(),
    restaurants -> restaurants.stream().filter(restaurant -> restaurant.id == idRestaurant).findFirst().orElse(null)
)